diff --git a/Makefile.am b/Makefile.am
index 060729642..d5e488b55 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,7 +8,7 @@ SUBDIRS += rpm
 endif

 if CONFIG_USER
-SUBDIRS += man scripts lib tests cmd etc contrib
+SUBDIRS += man scripts lib cmd etc contrib
 if BUILD_LINUX
 SUBDIRS += udev
 endif
@@ -125,7 +125,7 @@ cstyle:

 filter_executable = -exec test -x '{}' \; -print

-SHELLCHECKDIRS = cmd contrib etc scripts tests
+SHELLCHECKDIRS = cmd contrib etc scripts
 SHELLCHECKSCRIPTS = autogen.sh

 PHONY += checkabi storeabi
