diff --git a/Makefile.am b/Makefile.am
index 060729642..d5e488b55 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,7 +8,7 @@ SUBDIRS += rpm
 endif
 
 if CONFIG_USER
-SUBDIRS += man scripts lib tests cmd etc contrib
+SUBDIRS += man scripts lib cmd etc contrib
 if BUILD_LINUX
 SUBDIRS += udev
 endif
@@ -125,7 +125,7 @@ cstyle:
 
 filter_executable = -exec test -x '{}' \; -print
 
-SHELLCHECKDIRS = cmd contrib etc scripts tests
+SHELLCHECKDIRS = cmd contrib etc scripts
 SHELLCHECKSCRIPTS = autogen.sh
 
 PHONY += checkabi storeabi
diff --git a/configure.ac b/configure.ac
index 6f34b210d..f7f10f49d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -194,220 +194,6 @@ AC_CONFIG_FILES([
 	rpm/redhat/zfs-kmod.spec
 	rpm/redhat/zfs.spec
 	scripts/Makefile
-	tests/Makefile
-	tests/runfiles/Makefile
-	tests/test-runner/Makefile
-	tests/test-runner/bin/Makefile
-	tests/test-runner/include/Makefile
-	tests/test-runner/man/Makefile
-	tests/zfs-tests/Makefile
-	tests/zfs-tests/callbacks/Makefile
-	tests/zfs-tests/cmd/Makefile
-	tests/zfs-tests/cmd/badsend/Makefile
-	tests/zfs-tests/cmd/btree_test/Makefile
-	tests/zfs-tests/cmd/chg_usr_exec/Makefile
-	tests/zfs-tests/cmd/devname2devid/Makefile
-	tests/zfs-tests/cmd/draid/Makefile
-	tests/zfs-tests/cmd/dir_rd_update/Makefile
-	tests/zfs-tests/cmd/file_check/Makefile
-	tests/zfs-tests/cmd/file_trunc/Makefile
-	tests/zfs-tests/cmd/file_write/Makefile
-	tests/zfs-tests/cmd/get_diff/Makefile
-	tests/zfs-tests/cmd/largest_file/Makefile
-	tests/zfs-tests/cmd/libzfs_input_check/Makefile
-	tests/zfs-tests/cmd/mkbusy/Makefile
-	tests/zfs-tests/cmd/mkfile/Makefile
-	tests/zfs-tests/cmd/mkfiles/Makefile
-	tests/zfs-tests/cmd/mktree/Makefile
-	tests/zfs-tests/cmd/mmap_exec/Makefile
-	tests/zfs-tests/cmd/mmap_libaio/Makefile
-	tests/zfs-tests/cmd/mmapwrite/Makefile
-	tests/zfs-tests/cmd/nvlist_to_lua/Makefile
-	tests/zfs-tests/cmd/randfree_file/Makefile
-	tests/zfs-tests/cmd/randwritecomp/Makefile
-	tests/zfs-tests/cmd/readmmap/Makefile
-	tests/zfs-tests/cmd/rename_dir/Makefile
-	tests/zfs-tests/cmd/rm_lnkcnt_zero_file/Makefile
-	tests/zfs-tests/cmd/send_doall/Makefile
-	tests/zfs-tests/cmd/stride_dd/Makefile
-	tests/zfs-tests/cmd/threadsappend/Makefile
-	tests/zfs-tests/cmd/user_ns_exec/Makefile
-	tests/zfs-tests/cmd/xattrtest/Makefile
-	tests/zfs-tests/include/Makefile
-	tests/zfs-tests/tests/Makefile
-	tests/zfs-tests/tests/functional/Makefile
-	tests/zfs-tests/tests/functional/acl/Makefile
-	tests/zfs-tests/tests/functional/acl/off/Makefile
-	tests/zfs-tests/tests/functional/acl/posix/Makefile
-	tests/zfs-tests/tests/functional/acl/posix-sa/Makefile
-	tests/zfs-tests/tests/functional/alloc_class/Makefile
-	tests/zfs-tests/tests/functional/arc/Makefile
-	tests/zfs-tests/tests/functional/atime/Makefile
-	tests/zfs-tests/tests/functional/bootfs/Makefile
-	tests/zfs-tests/tests/functional/btree/Makefile
-	tests/zfs-tests/tests/functional/cache/Makefile
-	tests/zfs-tests/tests/functional/cachefile/Makefile
-	tests/zfs-tests/tests/functional/casenorm/Makefile
-	tests/zfs-tests/tests/functional/channel_program/Makefile
-	tests/zfs-tests/tests/functional/channel_program/lua_core/Makefile
-	tests/zfs-tests/tests/functional/channel_program/synctask_core/Makefile
-	tests/zfs-tests/tests/functional/chattr/Makefile
-	tests/zfs-tests/tests/functional/checksum/Makefile
-	tests/zfs-tests/tests/functional/clean_mirror/Makefile
-	tests/zfs-tests/tests/functional/cli_root/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zdb/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_bookmark/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_change-key/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_clone/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_copies/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_create/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_destroy/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_diff/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_get/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_ids_to_path/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_inherit/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_jail/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_load-key/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_mount/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_program/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_promote/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_property/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_receive/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_rename/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_reservation/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_rollback/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_send/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_set/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_share/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_snapshot/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_sysfs/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_unload-key/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_unmount/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_unshare/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_upgrade/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zfs_wait/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_add/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_attach/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_clear/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_create/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_destroy/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_detach/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_events/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_expand/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_export/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_get/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_history/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_import/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_import/blockfiles/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_initialize/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_labelclear/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_offline/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_online/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_remove/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_reopen/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_replace/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_resilver/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_scrub/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_set/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_split/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_status/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_sync/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_trim/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_upgrade/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_upgrade/blockfiles/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_wait/Makefile
-	tests/zfs-tests/tests/functional/cli_root/zpool_wait/scan/Makefile
-	tests/zfs-tests/tests/functional/cli_user/Makefile
-	tests/zfs-tests/tests/functional/cli_user/misc/Makefile
-	tests/zfs-tests/tests/functional/cli_user/zfs_list/Makefile
-	tests/zfs-tests/tests/functional/cli_user/zpool_iostat/Makefile
-	tests/zfs-tests/tests/functional/cli_user/zpool_list/Makefile
-	tests/zfs-tests/tests/functional/cli_user/zpool_status/Makefile
-	tests/zfs-tests/tests/functional/compression/Makefile
-	tests/zfs-tests/tests/functional/cp_files/Makefile
-	tests/zfs-tests/tests/functional/crtime/Makefile
-	tests/zfs-tests/tests/functional/ctime/Makefile
-	tests/zfs-tests/tests/functional/deadman/Makefile
-	tests/zfs-tests/tests/functional/delegate/Makefile
-	tests/zfs-tests/tests/functional/devices/Makefile
-	tests/zfs-tests/tests/functional/events/Makefile
-	tests/zfs-tests/tests/functional/exec/Makefile
-	tests/zfs-tests/tests/functional/fallocate/Makefile
-	tests/zfs-tests/tests/functional/fault/Makefile
-	tests/zfs-tests/tests/functional/features/Makefile
-	tests/zfs-tests/tests/functional/features/async_destroy/Makefile
-	tests/zfs-tests/tests/functional/features/large_dnode/Makefile
-	tests/zfs-tests/tests/functional/grow/Makefile
-	tests/zfs-tests/tests/functional/history/Makefile
-	tests/zfs-tests/tests/functional/hkdf/Makefile
-	tests/zfs-tests/tests/functional/inheritance/Makefile
-	tests/zfs-tests/tests/functional/inuse/Makefile
-	tests/zfs-tests/tests/functional/io/Makefile
-	tests/zfs-tests/tests/functional/l2arc/Makefile
-	tests/zfs-tests/tests/functional/large_files/Makefile
-	tests/zfs-tests/tests/functional/largest_pool/Makefile
-	tests/zfs-tests/tests/functional/libzfs/Makefile
-	tests/zfs-tests/tests/functional/limits/Makefile
-	tests/zfs-tests/tests/functional/link_count/Makefile
-	tests/zfs-tests/tests/functional/log_spacemap/Makefile
-	tests/zfs-tests/tests/functional/migration/Makefile
-	tests/zfs-tests/tests/functional/mmap/Makefile
-	tests/zfs-tests/tests/functional/mmp/Makefile
-	tests/zfs-tests/tests/functional/mount/Makefile
-	tests/zfs-tests/tests/functional/mv_files/Makefile
-	tests/zfs-tests/tests/functional/nestedfs/Makefile
-	tests/zfs-tests/tests/functional/no_space/Makefile
-	tests/zfs-tests/tests/functional/nopwrite/Makefile
-	tests/zfs-tests/tests/functional/online_offline/Makefile
-	tests/zfs-tests/tests/functional/pam/Makefile
-	tests/zfs-tests/tests/functional/pool_checkpoint/Makefile
-	tests/zfs-tests/tests/functional/pool_names/Makefile
-	tests/zfs-tests/tests/functional/poolversion/Makefile
-	tests/zfs-tests/tests/functional/privilege/Makefile
-	tests/zfs-tests/tests/functional/procfs/Makefile
-	tests/zfs-tests/tests/functional/projectquota/Makefile
-	tests/zfs-tests/tests/functional/pyzfs/Makefile
-	tests/zfs-tests/tests/functional/quota/Makefile
-	tests/zfs-tests/tests/functional/raidz/Makefile
-	tests/zfs-tests/tests/functional/redacted_send/Makefile
-	tests/zfs-tests/tests/functional/redundancy/Makefile
-	tests/zfs-tests/tests/functional/refquota/Makefile
-	tests/zfs-tests/tests/functional/refreserv/Makefile
-	tests/zfs-tests/tests/functional/removal/Makefile
-	tests/zfs-tests/tests/functional/rename_dirs/Makefile
-	tests/zfs-tests/tests/functional/replacement/Makefile
-	tests/zfs-tests/tests/functional/reservation/Makefile
-	tests/zfs-tests/tests/functional/rootpool/Makefile
-	tests/zfs-tests/tests/functional/rsend/Makefile
-	tests/zfs-tests/tests/functional/scrub_mirror/Makefile
-	tests/zfs-tests/tests/functional/slog/Makefile
-	tests/zfs-tests/tests/functional/snapshot/Makefile
-	tests/zfs-tests/tests/functional/snapused/Makefile
-	tests/zfs-tests/tests/functional/sparse/Makefile
-	tests/zfs-tests/tests/functional/suid/Makefile
-	tests/zfs-tests/tests/functional/threadsappend/Makefile
-	tests/zfs-tests/tests/functional/tmpfile/Makefile
-	tests/zfs-tests/tests/functional/trim/Makefile
-	tests/zfs-tests/tests/functional/truncate/Makefile
-	tests/zfs-tests/tests/functional/upgrade/Makefile
-	tests/zfs-tests/tests/functional/user_namespace/Makefile
-	tests/zfs-tests/tests/functional/userquota/Makefile
-	tests/zfs-tests/tests/functional/vdev_zaps/Makefile
-	tests/zfs-tests/tests/functional/write_dirs/Makefile
-	tests/zfs-tests/tests/functional/xattr/Makefile
-	tests/zfs-tests/tests/functional/zpool_influxdb/Makefile
-	tests/zfs-tests/tests/functional/zvol/Makefile
-	tests/zfs-tests/tests/functional/zvol/zvol_ENOSPC/Makefile
-	tests/zfs-tests/tests/functional/zvol/zvol_cli/Makefile
-	tests/zfs-tests/tests/functional/zvol/zvol_misc/Makefile
-	tests/zfs-tests/tests/functional/zvol/zvol_swap/Makefile
-	tests/zfs-tests/tests/perf/Makefile
-	tests/zfs-tests/tests/perf/fio/Makefile
-	tests/zfs-tests/tests/perf/regression/Makefile
-	tests/zfs-tests/tests/perf/scripts/Makefile
-	tests/zfs-tests/tests/stress/Makefile
 	udev/Makefile
 	udev/rules.d/Makefile
 	zfs.release
diff --git a/rpm/generic/zfs.spec.in b/rpm/generic/zfs.spec.in
index 7fa28ac72..8b4a105dd 100644
--- a/rpm/generic/zfs.spec.in
+++ b/rpm/generic/zfs.spec.in
@@ -533,9 +533,6 @@ systemctl --system daemon-reload >/dev/null || true
 %doc AUTHORS COPYRIGHT LICENSE NOTICE README.md
 
 %files test
-%{_datadir}/%{name}/zfs-tests
-%{_datadir}/%{name}/test-runner
-%{_datadir}/%{name}/runfiles
 %{_datadir}/%{name}/*.sh
 
 %files dracut
